lc-hax/Scripts/Commands/SellCommand.cs

```diff
@@ -16,12 +19,68 @@ public class SellCommand : ICommand {
             return;
         }
 
-        HaxObjects.Instance?.GrabbableObjects.ForEach(nullableGrabbableObject => {
-            if (!nullableGrabbableObject.IsNotNull(out GrabbableObject grabbableObject)) return;
-            if (!grabbableObject.itemProperties.isScrap) return;
+        if (args.Length is 0) {
+            SellEverything();
+            return;
+        }
+
+        if (args.Length is 1 && int.TryParse(args[0], out int targetAmount)) {
+            SellScrapAmount(targetAmount);
+            return;
+        }
+
+        void SellEverything() {
+            HaxObjects.Instance?.GrabbableObjects.ForEach(nullableGrabbableObject => {
+                if (!nullableGrabbableObject.IsNotNull(out GrabbableObject grabbableObject)) return;
+                if (!grabbableObject.itemProperties.isScrap) return;
+                if (grabbableObject.itemProperties.isDefensiveWeapon) return;
+                if (grabbableObject.isHeld) return;
+                SellObject(grabbableObject);
+            });
+        }
+
+        void SellScrapAmount(int value) {
+            if (value <= 0) {
+                Chat.Print("You must specify a positive amount of scrap to sell!");
+                return;
+            }
+
+            List<GrabbableObject?>? scraps = HaxObjects.Instance?.GrabbableObjects.Objects
+                .Where(nullableGrabbableObject =>
+                    nullableGrabbableObject.IsNotNull(out GrabbableObject grabbableObject) &&
+                    grabbableObject.itemProperties.isScrap && !grabbableObject.itemProperties.isDefensiveWeapon && !grabbableObject.isHeld
+                )
+                .OrderByDescending(grabbableObject => grabbableObject != null ? grabbableObject.scrapValue : 0)
+                .ToList();
+
+            System.Random random = new();
+            int remainingAmount = value;
+
+            while (scraps != null && remainingAmount > 0 && scraps.Count > 0) {
+                int randomIndex = random.Next(0, scraps.Count);
+                GrabbableObject? selectedScrap = scraps[randomIndex];
+
+                if (selectedScrap != null) {
+                    float discountedScrapValue = selectedScrap.scrapValue * Helper.StartOfRound.companyBuyingRate;
+
+                    SellObject(selectedScrap);
+
+                    remainingAmount -= (int)discountedScrapValue;
+                }
+
+                scraps.RemoveAt(randomIndex);
+            }
+
+            Chat.Print(remainingAmount <= 0
+                ? $"Sold scraps to reach or exceed the target amount of {value}!"
+                : $"Could not sell enough scraps to reach the target amount. Remaining scrap: {remainingAmount}");
+        }
+
 
-            player.currentlyHeldObjectServer = grabbableObject;
+        void SellObject(GrabbableObject item) {
+            if (item == null || player == null || depositItemsDesk == null) return;
```

Comment:
Why is this null check necessary when you assigned `item` as a non-nullable type?

Reply:
Just a habit i guess